### PR TITLE
docs(rxForm): FRMW-177-Migrate rxForm to API Docs

### PR DESCRIPTION
**Pull request by 100stacks**
JIRA: https://jira.rax.io/browse/FRMW-177

- [x] Dev LGTM
- [x] Dev LGTM

Migrate `rxForm` to API Documentation.

### CODE REVIEW

**Comment by bry4n** [comment]
LGTM

File: src/rxForm/rxForm.js [hunk 1]
```diff
@@ -4,25 +4,207 @@
...
+ * 
+ * ### Monthly Cost
+ * #### Before
+ * <pre>
+ * <form name="demoForm">
+ *   <rx-form-item label="Monthly Cost" prefix="$" suffix="million">
+ *     <input type="number" ng-model="volume.cost" />
+ *   </rx-form-item>
+ * </form>
+ * <pre>
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: Should be closing tag: `</pre>`

File: src/rxForm/rxForm.js [hunk 2]
```diff
@@ -670,6 +852,13 @@ angular.module('encore.ui.rxForm', ['ngSanitize', 'encore.ui.rxMisc'])
  * @description
  * Set of utility functions used by rxForm to access form data
  *
+ * # Services
+ * ## rxFormUtils
+ * Provides a set of utility functions for accessing form data.
+ * 
+ * **NOTE:** Nothing changed with this service in the update. You may continue to use this service as usual.*
+ * 
+ *
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: Existing documentation will suffice. This additional content is unnecessary. Please remove.

File: src/rxForm/rxForm.js [hunk 3]
```diff
@@ -4,25 +4,207 @@
...
+ * 
+ * # Layout
+ * 
+ * ## Stacked Field Arrangement
+ * By default, `rx-form-section` will arrange its children inline, in a row.  To obtain a stacked, columnar layout
+ * for a particular section, place the `stacked` attribute on the `rx-form-section` element.  This will arrange the
+ * `rx-field` and `div` children elements in a columnar fashion.  This can be used in conjunction with sections taking
+ *  the full width of the form.
+ *
+ *  *See "Advanced Inputs" in the demo for an example.*
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: I would make "demo" a link back to the component page.

```
{@link /#/components/rxForm demo}
```

File: src/rxForm/rxForm.js [hunk 4]
```diff
@@ -4,25 +4,207 @@
...
+ *
+ * ## Responsive
+ * `rx-field` and `div` elements that are immediate children of `rx-form-section` will grow from 250px to full width of 
+ * the section.  As such, you will see that these elements will wrap in the section if there's not enough width to 
+ * accomodate more than one child.
+ * 
+ * *You can see this in sections below if you resize the width of your browser.*
```
Comment (CITguy) [inline src/rxForm/rxForm.js]:
```
*You can see this in the {@link /#/components/rxForm demo} if you resize the width of your browser.*
```

File: src/rxForm/rxForm.js [hunk 5]
```diff
@@ -4,25 +4,207 @@
...
+ *
+ * # Validation
+ *
+ * ## Required Fields
+ * When displaying a field that should be required, please make use of the `ng-required` attribute for rxFieldName.  
+ * When the value evaluates to true, an asterisk will display to the left of the field name.  You can see an example 
+ * of this with the "Required Textare" field name in the demo.
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: * `Textarea` misspelled
* make "demo" link

```
of this with the "Required Textarea" field name in the {@link /#/components/rxForm demo}
```

File: src/rxForm/rxForm.js [hunk 6]
```diff
@@ -4,25 +4,207 @@
...
+ * 
+ * ## Custom Validators
+ *
+ * Angular provides its own validator when you use `type="email"`, and you can use `<rx-inline-error>` to turn email
+ * validation errors into a styled message.  You can also use this element if you define a custom validator.
+ *
+ * ### Foocheck validator
+ * The example shown in the "Email Address" example, uses a custom `foocheck` validator. Note that it is enabled by 
+ * placing the `foocheck` attribute in the `<input>` element, and using it with
+ * `ng-show="demoForm.userEmail.$error.foocheck"`.  Check out the Javascript tab in the demo below to see how this
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: * make "demo" link
* modify wording to reference demo page

```
`ng-show="demoForm.userEmail.$error.foocheck"`.  Check out the Javascript tab in the {@link /#/components/rxForm demo} to see how this
```

File: src/rxForm/rxForm.js [hunk 7]
```diff
@@ -4,25 +4,207 @@
...
+ *
+ * ### **rxFormOptionTable**
+ * Please use {@link rxOptionTable} as a stand-in replacement.
+ *
+ * ### **rxFormItem**
+ * See "Before & After" below
+ *
+ * ### **rxFormFieldset**
+ * Closest equivalent is to use `rxFormSection`. Your individual project requirements will vary, but the `legend` 
+ * attribute can be replaced with a `.title` heading variant where applicable.
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: * remove reference to `.title`

```
attribute can be replaced with a heading variant where applicable.
```

File: src/rxForm/rxForm.js [hunk 8]
```diff
@@ -4,25 +4,207 @@
...
+ *
+ * * If your legend pertains to at least one row, place a `.title` heading variant before the desired `rx-form-section`
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: * remove reference to `.title`

```
* If your legend pertains to at least one row, place a heading before the desired `rx-form-section`
```

File: src/rxForm/rxForm.js [hunk 9]
```diff
@@ -4,25 +4,207 @@
...
+ * element.
+ * * If your legend pertains to controls in a single column, place a `.title` heading variant within the `rx-field`
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: * remove reference to `.title`

```
* If your legend pertains to controls in a single column, place a heading within the `rx-field`
```

File: src/rxForm/rxForm.js [hunk 10]
```diff
@@ -4,25 +4,207 @@
  * @description
  * # rxForm Component
  *
- * rxForm components are a set of directives used to create forms throughout Encore.
- * These directives provide a common HTML layout and style for all form elements,
- * which helps ensure form accessibility and makes creating new forms easier.
+ * `rxForm` components are a set of directives used to create forms throughout Encore.  These directives provide a
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: * Do not use backticks for rxForm, instead start with `The rxForm component...`
* correct grammar

```
The rxForm component is a set of directives used to create forms throughout Encore.  These directives provide a
```

**Comment by 100stacks** [comment]
@CITguy sounds good on the updates.:ok_hand: :+1:   I just pushed the changes

File: src/rxForm/rxForm.js [hunk 11]
```diff
@@ -4,25 +4,207 @@
  * @description
  * # rxForm Component
  *
- * rxForm components are a set of directives used to create forms throughout Encore.
- * These directives provide a common HTML layout and style for all form elements,
- * which helps ensure form accessibility and makes creating new forms easier.
+ * The rxForm component is a set of directives used to create forms throughout Encore.  These directives provide a
```
Comment (CITguy) [inline src/rxForm/rxForm.js]: I'd go with `The rxForm component contains a set of directives...`

Comment (100stacks) [inline src/rxForm/rxForm.js]: Okay fixed.  :+1:

**Comment by CITguy** [comment]
LGTM

@100stacks, please squash

**Pull request merged**